src/settings/plugins/keyfile/nms-keyfile-plugin.c

```diff
@@ -533,16 +554,16 @@ add_connection (NMSettingsPlugin *config,
 	gs_free char *path = NULL;
 	gs_unref_object NMConnection *reread = NULL;
 
-	if (save_to_disk) {
-		if (!nms_keyfile_writer_connection (connection,
-		                                    NULL,
-		                                    FALSE,
-		                                    &path,
-		                                    &reread,
-		                                    NULL,
-		                                    error))
-			return NULL;
-	}
+	if (!nms_keyfile_writer_connection (connection,
+	                                    save_to_disk,
+	                                    NULL,
+	                                    FALSE,
+	                                    &path,
+	                                    &reread,
+	                                    NULL,
+	                                    error))
+		return NULL;
+
```

Comment:
I wonder, this works only if the user doesn't have another settings plugin enabled, right?
If, for example, user has 'plugins=ifcfg-rh' in the configuration, the keyfile add_connection() is not called and the connection file is not written into /run. Is this by design?

Reply:
No, ifcfg-rh does too. Though, that maybe should be avoided: https://bugzilla.gnome.org/show_bug.cgi?id=744711

Reply:
In any case, I think the current behavior is good. I don't think we want the ifcfg plugin to read or write connections in /run.